### ISSUE TITLE
Pydantic cannot generate JSON Schema for ObjectIdField object

### ISSUE THREAD

**Issue by ensaremirerol** [jefersondaniel/pydantic-mongo#83]
There isn't much to explain. It seems that changes introduced in Pydantic 2.4 have broken this object, and when you try to use it with FastAPI, it throws the following exception:
```
pydantic.errors.PydanticInvalidForJsonSchema: 
Cannot generate a JsonSchema for core_schema.PlainValidatorFunctionSchema 
({'type': 'no-info', 
'function': <bound method ObjectIdField.validate of <class 'ObjectIdField'>>})
```

The solution is simple: using only str_validator on json_schema solves the issue.

If you are interested in proceeding with this solution, please let me know so I can create a pull request

**Comment by ensaremirerol** [comment]
@Rekoc you can use my fork until this PR accepted

**Comment by jefersondaniel** [comment]
The PR is merged and available in the version 2.1.2
